File: accsaber-database/src/main/java/de/ixsen/accsaber/database/repositories/model/CategoryRepository.java
package de.ixsen.accsaber.database.repositories.model;

import de.ixsen.accsaber.database.model.Category;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface CategoryRepository extends JpaRepository<Category, String> {

    boolean existsByCategoryName(String categoryName);

    Optional<Category> findByCategoryName(String categoryName);

    void deleteByCategoryName(String categoryName);
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/repositories/model/CategoryRepository.kt
package de.ixsen.accsaber.database.repositories.model

import de.ixsen.accsaber.database.model.Category
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CategoryRepository : JpaRepository<Category, String> {
    fun existsByCategoryName(categoryName: String): Boolean
    fun findByCategoryName(categoryName: String): Optional<Category>
    fun deleteByCategoryName(categoryName: String)
}